docs/001_develop/03_client-capabilities/014_utility-methods/12_styles-util.mdx

```diff
@@ -0,0 +1,115 @@
+---
+title: 'Styles utilities'
+sidebar_label: 'Styles utilities'
+id: styles-util
+keywords: [utils, utility, styles, dynamic, look, styling, management]
+tags:
+  - utils
+  - utility
+  - styles
+  - dynamic
+  - look
+  - styling
+  - management
+---
+
+This  encompasses a comprehensive suite of utilities and components for dynamic style management, including CSS rule insertion, font face loading, and the application of styles to custom elements.
+
+## Key Features
+
+- **Encapsulated Style Management:** Allows styles to be defined and dynamically changed for custom elements, supporting encapsulated styling strategies.
+- **Font Face Loading:** Simplifies the loading of custom font faces by generating and applying the necessary CSS rules.
+- **Dynamic CSS Rule Insertion:** Enables the insertion of CSS rules into the document, facilitating dynamic style updates.
+- **Element Style Conversion:** Converts `ComposableStyles` or an array of `ComposableStyles` into `ElementStyles` for efficient style application.
+
+## Use Cases
+
+This module is ideal for:
+- Dynamic style management.
+- Enhanced styling, flexibility and maintainability
+
+### Examples
+
+#### Dynamic Style Application with `SlottedStyles`
+
+The `SlottedStyles` custom element enables the encapsulation and dynamic application of styles to slotted content. It observes changes to its `styles` property and applies the new styles to its parent element's shadow DOM or to the document.
+
+```typescript
+import { SlottedStyles } from '@genesislcap/foundation-utils';
+
+// Create a new SlottedStyles element
+const slottedStyles = document.createElement('slotted-styles') as SlottedStyles;
+
+// Define styles
+slottedStyles.styles = `
+  :host {
+    display: block;
+    color: red;
+  }
+`;
```

Comment:
should the example for this component be 

`
<rapid-button>
  Text will be red
  <slotted-styles styles={ display: block, color: red}></slotted-styles>
</rapid-button>
`

Reply:
So, testing in fondation UI, angular and react.

Define styles
`const slottedStyles = css`
  :host {
    display: block;
    color: red;
  }
`;
`

In angular

`        <rapid-button>
            <slotted-styles [styles]="slottedStyles"></slotted-styles>
            Slotted Styles button
        </rapid-button>`

In foudnation

`
          <rapid-button>
            <slotted-styles :styles="${() => slottedStyles}"></slotted-styles>
            Slotted styles button
          </rapid-button>
`

In react, does not work for me without use ref

`
  useEffect(() => {

    if (slottedStylesRef.current) {
      slottedStylesRef.current.styles = slottedStyles;
    }
    
    });
    
    return (
            <rapid-button>
          <slotted-styles ref={slottedStylesRef}></slotted-styles>
          Slotted Styles button
        </rapid-button>
    )
`